platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -177,13 +126,9 @@ static void mbed_minimal_putchar(char *buffer, size_t length, int *result, char
             if (buffer) {
                 buffer[*result] = data;
             } else {
-#if MBED_CONF_PLATFORM_MINIMAL_PRINTF_ENABLE_FILE_STREAM
+                /* print character to stdout or file */
```

Comment:
Think the `if` is redundant? `stream` must be set, or this does nothing.

I note now that this means there's no error-handling for `printf`. Seems a reasonable minor optimisation, but might be worth adding to the documentation if it's not there already.

Reply:
`stream` should be set. I left the check for NULL for defensive prorgamming.

Reply:
Not the sort of defensive programming that's helpful - you don't really want to "quietly do nothing" if passed NULL, you want to find out who did it and slap them. Standard printf certainly wouldn't put up with it.

I agree that being defensive and coping with NULL in a `%s` parameter can be helpful, when you've got millions of people passing you strings, many of them already in debug paths, but extending it to the `FILE *` takes it a bit too far IMO. And `%s` would at least print "<null>", not do nothing.

(Although you're not actually being defensive on the null '%s'!)